src/App.tsx

```diff
@@ -14,11 +16,100 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  selectedGood: string[],
+};
+
+type Props = {};
+
+export class App extends React.Component<Props, State> {
+  state = {
+    selectedGood: ['Jam'],
+  };
+
+  goodsArr = this.state.selectedGood;
+
+  addGood = (good: string) => {
+    this.setState((state) => ({
+      selectedGood: [...state.selectedGood, good],
+    }));
+  };
+
+  removeGood = (good: string) => {
+    this.setState((state: State) => {
+      state.selectedGood.splice(state.selectedGood.indexOf(good), 1);
+
+      return { selectedGood: state.selectedGood };
+    });
+  };
+
+  cleanBin = () => this.setState({ selectedGood: [] });
+
+  textMaker = (goodsArr: string[]) => {
+    if (goodsArr.length === 1) {
+      return ` ${goodsArr[0]} is selected.`;
+    }
+
+    if (goodsArr.length === 2) {
+      return `${goodsArr[0]} and ${goodsArr[1]} are selected.`;
+    }
+
+    if (goodsArr.length > 2) {
+      let str = goodsArr[0];
+
+      for (let i = 1; i < goodsArr.length - 1; i += 1) {
+        str += `, ${goodsArr[i]}`;
+      }
+
+      return `${str} and ${goodsArr[goodsArr.length - 1]} are selected.`;
+    }
+
+    return 'No goods selected.';
+  };
 
-export default App;
+  render() {
+    return (
+      <div className="App">
+        <div className="App__bin">
+          <h1>
+            {this.textMaker(this.state.selectedGood)}
+          </h1>
+          {this.state.selectedGood.length > 0
+          && (
+            <button
+              type="button"
+              className="App__cleanBin"
+              onClick={() => this.cleanBin()}
+            >
+              Clean Bin
+            </button>
+          )}
+        </div>
+        <ul className="App__list">
+          {goodsFromServer.map(good => {
+            return (
+              <li key={good} className="App__goodBox">
+                <span>{good}</span>
+                <button
+                  onClick={
+                    this.state.selectedGood.includes(good)
```

Comment:
here you can check includes one time and after use isIncludesGood

```suggestion
          {goodsFromServer.map(good => {
            const isIncludesGood = this.state.selectedGood.includes(good);
     
            return (
              <li key={good} className="App__goodBox">
                <span>{good}</span>
                <button
                  onClick={
                    isIncludesGood
```